controllers/userController.js

```diff
@@ -434,17 +485,30 @@ function parseErrors(errors) {
 
  			mailer.send(mail.to, mail.subject, mail.text)
  			.then(function() {
- 				req.flash('info', {
- 					message: 'An e-mail has been sent to ' + user.email + ' with further instructions.'
- 				})
+				if (!wantJSON) {
+					req.flash('info', {
+						message: 'We emailed further instructions to ' + user.email + '.'
+					})
+				}
  				done()
  			})
  			.catch(done)
  		}
  		], function(err) {
- 			if (err)
+ 			if (err) {
+				res.status(500).json({ok:false, message: 'The server could not email you. Please contact us for assistance.'})
  				return next(err)
+			}
+			if (wantJSON) {
+				return res.status(200).json({
+					ok:		true,
```

Comment:
also since these should be the same for every response, consider making these functions, something like:
```
var _ = require('lodash')
function ResponseStatusSuccess(message, options) {
    return _.extend(options || {}, { success: true, message: message})
}

```

that way it's easier to change them all at once